cpu/native/netdev_raw802154/netdev_raw802154.c

```diff
@@ -0,0 +1,337 @@
+                return -EOVERFLOW;
+            }
+            *((uint16_t *)val) = _get_txpower(dev);
+            return sizeof(uint16_t);
+            break;
+        case NETOPT_CSMA:
+            if (max_len < sizeof(netopt_enable_t)) {
+                return -EOVERFLOW;
+            }
+            *((netopt_enable_t *)val) = NETOPT_DISABLE;
+            return sizeof(netopt_enable_t);
+        case NETOPT_AUTOCCA:
+            if (max_len < sizeof(netopt_enable_t)) {
+                return -EOVERFLOW;
+            }
+            *((netopt_enable_t *)val) = NETOPT_DISABLE;
+            return sizeof(netopt_enable_t);
+        case NETOPT_RETRANS:
+            if (max_len < sizeof(uint8_t)) {
+                return -EOVERFLOW;
+            }
+            *((uint8_t *)val) = _get_max_retries(dev);
+            return sizeof(uint8_t);
+            break;
+        default:
+            break;
+    }
+    int res;
+    if (((res = netdev_ieee802154_get((netdev_ieee802154_t *)netdev, opt, val,
+                                       max_len)) >= 0) || (res != -ENOTSUP)) {
+        return res;
+    }
+    return 0;
+}
+
+static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
+{
+    return netdev_ieee802154_set((netdev_ieee802154_t *)netdev, opt,
+                                  val, len);
+}
+
+static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
+{
+    netdev_raw802154_t *dev = (netdev_raw802154_t *)netdev;
+    (void)info;
+    int nread;
+
+    if (!buf) {
+        if (len > 0)
+            native_async_continue_reading(dev->fd);
+        else
+            return 127;
+    }
+
+    nread = real_read(dev->fd, buf, len) - 2;
+    if (nread > 0) {
+        native_async_continue_reading(dev->fd);
+
+#ifdef MODULE_NETSTATS_L2
+        netdev->stats.rx_count++;
+        netdev->stats.rx_bytes += nread;
+#endif
+    }
+
+    return nread;
+}
+
+static int _send(netdev_t *netdev, const iolist_t *iolist)
+{
+    netdev_raw802154_t *dev = (netdev_raw802154_t *)netdev;
+    struct iovec iov[iolist_count(iolist)];
+    size_t bytes;
+    unsigned n;
+
+    bytes = iolist_to_iovec(iolist, iov, &n);
+
+    int res = _native_writev(dev->fd, iov, n);
+#ifdef MODULE_NETSTATS_L2
+    netdev->stats.tx_bytes += bytes;
+#else
+    (void)bytes;
+#endif
+    if (netdev->event_callback)
+        netdev->event_callback(netdev, NETDEV_EVENT_TX_COMPLETE);
+
+    return res;
+}
+
+static int _init(netdev_t *netdev)
+{
+    netdev_raw802154_t *dev = (netdev_raw802154_t *)netdev;
+    uint64_t extended_rand_addr;
+    struct sockaddr_ll sll;
+    struct ifreq ifr;
+    int ret;
+
+#ifdef MODULE_NETSTATS_L2
+    memset(&netdev->stats, 0, sizeof(netstats_t));
+#endif
+
+    /* reset options and sequence number */
+    dev->netdev.seq = 0;
+
+    _set_page(dev, 0);
+    _set_chan(dev, 11);
+
+    if (dev->rand_addr) {
+        hwrng_init();
+        hwrng_read((uint8_t *)&extended_rand_addr, sizeof(extended_rand_addr));
```

Comment:
Mhhh is there a way (like with `netdev_tap`) to have an address that is the same every time when restarting the `native` routine apart from a predefined macro? Apart from making it easier on the user, that would also allow to remove the need for `hwrng_init()` modifications in this PR.

Reply:
Mh... yeah, I just realized that for `netdev_tap` we basically just took the MAC address of the TAP +1 (this makes the address reusable when restarting the native process, since the TAP keeps existing [at least on Linux]), so I guess my earlier comment doesn't really make sense if you don't have that.